packages/data-service/src/import-export-connection.ts

```diff
@@ -0,0 +1,203 @@
+import { EJSON } from 'bson';
+import { cloneDeep } from 'lodash';
+import type { ConnectionInfo } from './connection-info';
+import type { ConnectionSecrets } from './connection-secrets';
+import { extractSecrets, mergeSecrets } from './connection-secrets';
+import { ConnectionStorage } from './connection-storage';
+import { Decrypter, Encrypter } from './encrypt';
+import createLoggerAndTelemetry from '@mongodb-js/compass-logging';
+
+const { log, mongoLogId, track } = createLoggerAndTelemetry(
+  'COMPASS-CONNECTION-IMPORT-EXPORT'
+);
+
+interface ConnectionInfoWithEncryptedData extends ConnectionInfo {
+  connectionSecrets?: string;
+}
+
+const kCurrentVersion = 1;
+const kFileTypeDescription = 'Compass Connections';
+
+export interface ExportImportConnectionOptions {
+  passphrase?: string;
+  filter?: (info: ConnectionInfo) => boolean;
+  storage?: Pick<ConnectionStorage, 'save' | 'loadAll'>;
+  trackingProps?: object;
+}
+
+export interface ExportConnectionOptions extends ExportImportConnectionOptions {
+  removeSecrets?: boolean;
+}
+
+export type ImportConnectionOptions = ExportImportConnectionOptions;
+
+export async function exportConnections(
+  options: ExportConnectionOptions = {}
+): Promise<string> {
+  const {
+    storage = options.storage ?? new ConnectionStorage(),
+    filter = (info) => info.favorite?.name,
+    passphrase = '',
+    removeSecrets = false,
+    trackingProps = {},
+  } = options;
+
+  if (passphrase && removeSecrets) {
+    throw new Error(
+      'Cannot both specify to remove secrets and provide a passphrase for encrypting secrets'
+    );
+  }
+
+  const allConnections = await storage.loadAll();
+  let exportConnections: ConnectionInfoWithEncryptedData[] = cloneDeep(
+    allConnections.filter(filter)
+  );
+  log.info(
+    mongoLogId(1_001_000_151),
```

Comment:
From where do we get these numbers? Is there somewhere a complete list with descriptions?

Reply:
There's no complete list. The number is incremental so each log gets its own id. There's a lint script to ensure we don't use the same one twice.

Reply:
So we just manually search in code which number is the last and add +1?

Reply:
`node scripts/check-logids.js` :)